eslint.config.mjs

```diff
@@ -737,132 +760,217 @@ const strictRules = {
-    'react-hooks',
-    '@typescript-eslint',
-    '@emotion',
-    'import',
-    'react',
-    'sentry',
-    'simple-import-sort',
-    'no-lookahead-lookbehind-regexp',
-  ],
-
-  parser: '@typescript-eslint/parser',
-
-  parserOptions: detectDeprecations
-    ? {
-        warnOnUnsupportedTypeScriptVersion: false,
-        ecmaVersion: 6,
-        sourceType: 'module',
-        ecmaFeatures: {
-          jsx: true,
-          modules: true,
-          legacyDecorators: true,
-        },
-        project: './tsconfig.json',
-      }
-    : {
-        warnOnUnsupportedTypeScriptVersion: false,
-        ecmaVersion: 6,
-        sourceType: 'module',
+// Used by both: `languageOptions` & `parserOptions`
+const ecmaVersion = 6; // TODO(ryan953): change to 'latest'
+
+/**
+ * To get started with this ESLint Configuration list be sure to read at least
+ * these sections of the docs:
+ *  - https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-and-ignores
+ *  - https://eslint.org/docs/latest/use/configure/configuration-files#cascading-configuration-objects
+ */
+export default typescript.config([
+  {
+    /**
+     * Main parser & linter options
+     * Rules are defined below and inherit these properties
+     * https://eslint.org/docs/latest/use/configure/configuration-files#configuration-objects
+     */
+    name: 'main',
+    languageOptions: {
+      ecmaVersion,
+      sourceType: 'module',
+      globals: {
+        // TODO(ryan953): globals.browser seems to have a bug with trailing whitespace
+        ...Object.fromEntries(
+          Object.keys(globals.browser).map(key => [key.trim(), false])
+        ),
+        ...globals.jest,
+        MockApiClient: true,
+        tick: true,
+      },
+      parser: typescript.parser,
+      parserOptions: {
         ecmaFeatures: {
-          jsx: true,
-          modules: true,
-          legacyDecorators: true,
+          globalReturn: false,
         },
-      },
+        ecmaVersion,
 
-  env: {
-    browser: true,
-    es6: true,
-    jest: true,
-  },
+        // https://typescript-eslint.io/packages/parser/#emitdecoratormetadata
+        emitDecoratorMetadata: undefined,
 
-  globals: {
-    require: false,
-    expect: false,
-    MockApiClient: true,
-    tick: true,
-    jest: true,
-  },
+        // https://typescript-eslint.io/packages/parser/#experimentaldecorators
+        experimentalDecorators: undefined,
+
+        // https://typescript-eslint.io/packages/parser/#jsdocparsingmode
+        jsDocParsingMode: 'none',
+
+        // https://typescript-eslint.io/packages/parser/#project
+        project: './tsconfig.json',
 
-  settings: {
-    react: {
-      version: '18.2.0', // React version, can not `detect` because of getsentry
+        // https://typescript-eslint.io/packages/parser/#projectservice
+        // `projectService` is slower with our current tsconfig files.
+        // projectService: true,
+        // tsconfigRootDir: import.meta.dirname,
```

Comment:
`projectService` is recommended, but it's much slower. With `project` runtime is about 2m30s and with `projectService` we're at 3m